investments/views.py

```diff
@@ -53,4 +61,19 @@ def create(self, request, *args, **kwargs):
             return Response(
                 {"detail": "Failed to create subscription. Please try again."},
                 status=status.HTTP_400_BAD_REQUEST,
-            )
\ No newline at end of file
+            )
+        
+    def perform_create(self, serializer):
+        """
+        Persist a new subscription instance.
+
+        - Retrieves the target project using the `project_id` from the URL.
+        - Associates the subscription with the authenticated investor (`request.user.investor`).
+        - Saves the subscription via the serializer.
+        """
+        project_id = self.kwargs["project_id"]
+        project = Project.objects.get(pk=project_id)
```

Comment:
Does not handle the Project.DoesNotExist exception, risking unhandled errors. Wrap the project fetch in try/except and raise a meaningful error.
```
try:
    project = Project.objects.get(pk=project_id)
except Project.DoesNotExist:
    raise ValidationError({"project": "Project does not exist."})
```